src/coreclr/runtime/portable/AllocFast.cpp

```diff
@@ -41,7 +50,64 @@ static Object* _RhpNewArrayFastCore(MethodTable* pMT, INT_PTR size)
         return pObject;
     }
 
-    return RhpNewVariableSizeObject(pMT, size);
+    return AllocateObject(pMT, 0, size);
+}
+
+static Object* NewArrayFastAlign8Core(MethodTable* pMT, INT_PTR size)
+{
+    FCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
+    _ASSERTE(size < INT32_MAX);
+
+    Thread* thread = GetThread();
+    ee_alloc_context* cxt = thread->GetEEAllocContext();
+
+    size_t sizeInBytes = (size_t)pMT->GetBaseSize() + ((size_t)size * (size_t)pMT->RawGetComponentSize());
+    sizeInBytes = ALIGN_UP(sizeInBytes, sizeof(void*));
+
+    uint8_t* alloc_ptr = cxt->getAllocPtr();
+    int requiresAlignObject = ((uint32_t)alloc_ptr) & 7;
+    size_t paddedSize = sizeInBytes;
+    if (requiresAlignObject)
+    {
+        paddedSize += 12;
+    }
+
+    _ASSERTE(alloc_ptr <= cxt->getAllocLimit());
+    if ((size_t)(cxt->getAllocLimit() - alloc_ptr) >= paddedSize)
+    {
+        cxt->setAllocPtr(alloc_ptr + paddedSize);
+        if (requiresAlignObject)
+        {
+            Object* dummy = (Object*)alloc_ptr;
+            dummy->SetMethodTable(g_pFreeObjectMethodTable);
+            alloc_ptr += 12;
```

Comment:
You can replace 12 with `MIN_OBJECT_SIZE` if you would like to avoid the magic constant.